### PR TITLE
fix: recover lost function arguments and argument types

### PR DESCRIPTION
**Pull request by dandhlee**
Before you open a pull request, note that this repository is forked from [here](https://github.com/docascode/sphinx-docfx-yaml/).
Unless the issue you're trying to solve is unique to this specific repository, 
please file an issue and/or send changes upstream to the original as well.

__________________________________________________________________

When there is a `*` argument present in the function signature, they are not included in `args` from `inspect.getfullargspec` function. Iterating through the entire list to make sure everything has been counted for will ensure all variables are accounted for.

As well, when there isn't a docstring entry for functions they only had their names (`id`) included in the table of parameters. Recovering their types if specified in the function signature, otherwise they'll be removed to have the tables not look so empty.

Fixes internally filed issue.

> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by dandhlee** [comment]
Ran a test Kokoro job, seems to work fine but it failed on `python-bigquery-sqlalchemy` which is expected - it currently produces no documentation for DocFX YAML.

File: docfx_yaml/extension.py [hunk 1]
```diff
@@ -517,22 +517,51 @@ def _update_friendly_package_name(path):
...
-                        args[len(args) - cut_count + count]['defaultValue'] = str(default)
+                # Attempt to add default values to arguments.
+                try:
+                    for count, default in enumerate(argspec.defaults):
+                        cut_count = len(argspec.defaults)
+                        # Only add defaultValue when str(default) doesn't contain object address string(object at 0x)
+                        # inspect.getargspec method will return wrong defaults which contain object address for some default values, like sys.stdout
+                        # Match the defaults with the count
+                        if 'object at 0x' not in str(default):
+                            args[len(args) - cut_count + count]['defaultValue'] = str(default)
```
Comment (tbpg) [inline docfx_yaml/extension.py:561]: I don't understand `len(args) - cut_count + count`. Can you clarify in a comment, different variable names, or something else?

File: docfx_yaml/extension.py [hunk 2]
```diff
@@ -643,12 +672,26 @@ def _update_friendly_package_name(path):
...
         for arg in args:
+            arg_id.append(arg['id'])
+
             if arg['id'] in variables:
                 # Retrieve argument info from extracted map of variable info
                 arg_var = variables[arg['id']]
                 arg['var_type'] = arg_var.get('var_type') if arg_var.get('var_type') else ''
                 arg['description'] = arg_var.get('description') if arg_var.get('description') else ''
+
+        # Add any variables we might have missed from extraction.
```
Comment (tbpg) [inline docfx_yaml/extension.py:685]: Do we need to worry about order of arguments?

File: docfx_yaml/extension.py [hunk 3]
```diff
@@ -643,12 +672,26 @@ def _update_friendly_package_name(path):
...
+        for variable in variables:
```
Comment (tbpg) [inline docfx_yaml/extension.py:686]: Could we have a single loop over `variables` and not `args`?

**Review by tbpg — CHANGES_REQUESTED** [review]
Any tests? :smile:

**Review by dandhlee — COMMENTED** [review]
Hard to make a specific unit test for the added bits as they need to be part of Sphinx with `app`, it would be easier to have this with the goldens, planning to fix it in #44.

**Review by tbpg — APPROVED** [review]

**Pull request merged**